gender_analysis/analysis/metadata_visualizations.py

```diff
@@ -36,39 +42,43 @@ def plot_pubyears(corpus, filename=None):
     sns.color_palette('colorblind')
     ax1 = plt.subplot2grid((1, 1), (0, 0))
     plt.figure(figsize=(10, 6))
-    bins = [num for num in range(min(pub_years), max(pub_years)+4, 5)]
+    bins = [ num for num in range(min(pub_years), max(pub_years) + 4, 5) ]
     plt.hist(pub_years, bins, histtype='bar', rwidth=.8, color='c')
     plt.xlabel('Year', size=15, weight='bold', color='k')
     plt.ylabel('Frequency', size=15, weight='bold', color='k')
-    plt.title('Publication Year Concentration for '+corpus_name.title(), size=18, weight='bold',
+    plt.title('Publication Year Concentration for ' + corpus_name.title(), size=18, weight='bold',
               color='k')
     plt.yticks(size=15, color='k')
-    plt.xticks([i for i in range(min(pub_years), max(pub_years)+9, 10)], size=15, color='k')
+    plt.xticks([ i for i in range(min(pub_years), max(pub_years) + 9, 10) ], size=15, color='k')
     for label in ax1.xaxis.get_ticklabels():
         label.set_rotation(60)
     plt.subplots_adjust(left=.1, bottom=.18, right=.95, top=.9)
 
+    created_file_path = generate_file_path_for_visualizations(file_path)
+    print(created_file_path)
     if filename:
-        plt.savefig(filename.replace(' ', '_')+'.png')
+        plt.savefig(str(created_file_path) + '/' + filename.replace(' ', '_') + '.png')
```

Comment:
Here and elsewhere below, please use `pathlib` to construct the path, rather than manually doing string manipulation. `pathlib` solves cross-platform issues: on Windows this `/` won't work.